srtcore/core.cpp

```diff
@@ -9721,59 +9721,56 @@ int srt::CUDT::handleSocketPacketReception(const vector<CUnit*>& incoming, bool&
                     uint64_t(m_stats.traceBelatedTime) * 1000,
                     count_microseconds(steady_clock::now() - pts), 0.2);
-
             m_stats.traceBelatedTime = bltime / 1000.0;
             m_stats.rcvr.recvdBelated.count(rpkt.getLength());
             leaveCS(m_StatsLock);
+
             HLOGC(qrlog.Debug,
-                    log << CONID() << "RECEIVED: seq=" << rpkt.m_iSeqNo << " offset=" << offset << " (BELATED/"
-                    << s_rexmitstat_str[pktrexmitflag] << ") FLAGS: " << rpkt.MessageFlagStr());
+                  log << CONID() << "RECEIVED: seq=" << rpkt.m_iSeqNo
+                      << " offset=" << CSeqNo::seqoff(m_iRcvLastSkipAck, rpkt.m_iSeqNo) << " (BELATED/"
+                      << s_rexmitstat_str[pktrexmitflag] << ") FLAGS: " << rpkt.MessageFlagStr());
             continue;
         }
 
-        const int avail_bufsize = (int) getAvailRcvBufferSizeNoLock();
+        IF_HEAVY_LOGGING(const char *exc_type = "EXPECTED");
+        bool adding_successful = true;
 
-        if (offset >= avail_bufsize)
+        int offset = 0;
+        int buffer_add_result = m_pRcvBuffer->insert(u, offset);
+        if (buffer_add_result == -3)
         {
-            // This is already a sequence discrepancy. Probably there could be found
-            // some way to make it continue reception by overriding the sequence and
-            // make a kinda TLKPTDROP, but there has been found no reliable way to do this.
+            // The insert() result is -3 if the insert offset exceeds capacity.
             if (m_bTsbPd && m_bTLPktDrop && m_pRcvBuffer->empty())
             {
+                // This is already a sequence discrepancy. Probably there could be found
+                // some way to make it continue reception by overriding the sequence and
+                // make a kinda TLKPTDROP, but there has been found no reliable way to do this.
                 // Only in live mode. In File mode this shall not be possible
                 // because the sender should stop sending in this situation.
                 // In Live mode this means that there is a gap between the
                 // lowest sequence in the empty buffer and the incoming sequence
                 // that exceeds the buffer size. Receiving data in this situation
                 // is no longer possible and this is a point of no return.
-
                 LOGC(qrlog.Error, log << CONID() <<
                         "SEQUENCE DISCREPANCY. BREAKING CONNECTION."
                         " seq=" << rpkt.m_iSeqNo
                         << " buffer=(" << m_iRcvLastSkipAck
                         << ":" << m_iRcvCurrSeqNo                   // -1 = size to last index
                         << "+" << CSeqNo::incseq(m_iRcvLastSkipAck, int(m_pRcvBuffer->capacity()) - 1)
-                        << "), " << (offset-avail_bufsize+1)
+                        << "), " << (offset - int(m_pRcvBuffer->capacity()) + 1)
```

Comment:
I don't like the `offset` being added to the `CRcvBuffer::insert(..)` only to print this log line.
Consider using `CRcvBuffer::getStartSeqNo()` instead.